### PR TITLE
introduce testing suite

### PR DESCRIPTION
**Pull request by KonradDanielewski**

### CODE REVIEW

File: plotlybrain/build_geoJSON.py [hunk 1]
```diff
@@ -741,49 +747,31 @@ def build_geojson(
...
     out_path: str,
-    lon_range: tuple[float, float] = (-15.0, 15.0),
-    lat_range: tuple[float, float] = (-10.0, 10.0),
 ) -> str:
     """
     Save a GeoJSON FeatureCollection to disk.
 
+    The geometry is written verbatim. ``build_geojson`` already scales
+    coordinates into pseudo lon/lat space, so this function must not scale
+    again (doing so would re-normalize and vertically flip the slice).
```
Comment (anna-teruel) [inline plotlybrain/build_geoJSON.py]: I agree with that, but you think we need to explain this on the docstrings?

File: tests/conftest.py [hunk 2]
```diff
@@ -0,0 +1,173 @@
+"""Shared fixtures for the PlotlyBrain backend test suite.
```
Comment (anna-teruel) [inline tests/conftest.py:1]: These fixtures look good to me. For now I don't come up with other ideas.

File: tests/test_build_geojson.py [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+def _flat_coords(geometry_coordinates):
+	return np.array(
+		[pt for poly in geometry_coordinates for ring in poly for pt in ring]
+	)
+
+
+def test_load_annotation_volume_rejects_unsupported_resolution():
+	# Validation happens before any network access.
+	with pytest.raises(ValueError):
+		load_annotation_volume(resolution_um=7)
```
Comment (anna-teruel) [inline tests/test_build_geojson.py:30]: Why the resolution is 7? I guess it's just a random low value?

File: tests/test_build_geojson.py [hunk 4]
```diff
@@ -0,0 +1,244 @@
+"""Tests for slice extraction, mask polygonization, and GeoJSON assembly."""
```
Comment (anna-teruel) [inline tests/test_build_geojson.py:1]: I think the main functionality is now covered. I added tests that directly cover ``scale_cartesian_to_lonlat()``, which was previously only tested indirectly through ``build_geojson()``, and a test for the coords_mm slice-selection path in ``build_geojson()``.

File: tests/test_choropleth_render.py [hunk 5]
```diff
@@ -0,0 +1,65 @@
+"""Tests for choropleth rendering and the value->color helper."""
```
Comment (anna-teruel) [inline tests/test_choropleth_render.py:1]: The choropleth rendering tests already cover the main behavior: figure creation, excluding default Allen IDs, missing feature_id errors, regions without scores, and the main ``value_to_color()`` edge cases. The only extra tests I would consider are small checks for ``zmin/zmax``, custom Plotly ``kwargs`` passed through ``**kwargs``. These are not blockers, but they would make the render-function coverage a bit more complete.

File: tests/test_coord_system.py [hunk 6]
```diff
@@ -0,0 +1,105 @@
+"""Tests for coordinate <-> slice-index conversions and range generation."""
```
Comment (anna-teruel) [inline tests/test_coord_system.py:1]: The only small thing I would consider adding is a test that ``range_mm_to_slice_indices()`` also works for ``sagittal`` or ``horizontal``, because most range tests currently rely on the default "coronal" orientation.

**Review by anna-teruel — COMMENTED** [review]
So far, I see that total coverage: 39%, because the Dash app is completely untested.
We can leave it as it is for now, but we need to think on how to test the Dash app?

One option would be to start by testing callback and figure generation functions directly with fixtures and mock inputs, and later add a small number of end-to-end tests that simulate user interactions (button clicks, dropdown selections, file loading, etc.) using Dash's testing framework. What do you think?

**Comment by anna-teruel** [comment]
I additionally removed my name from the scripts I wrote, because I saw you don't sign, so thought it was better to be consistent. It's here: 238c915213113ed21274b38aa93acba8884cb2cc

File: plotlybrain/build_geoJSON.py (hunk 1 above)
Comment (KonradDanielewski) [reply plotlybrain/build_geoJSON.py]: Ah, yeah we can ditch that at this point

File: tests/test_build_geojson.py (hunk 3 above)
Comment (KonradDanielewski) [reply tests/test_build_geojson.py:30]: Non existent resolution to check whether loading crashes correctly.

File: tests/test_coord_system.py (hunk 6 above)
Comment (KonradDanielewski) [reply tests/test_coord_system.py:1]: Generally we should take a look at other orientations. There is some weirdness in rendering for the horizontal view

**Comment by KonradDanielewski** [comment]
> I additionally removed my name from the scripts I wrote, because I saw you don't sign, so thought it was better to be consistent. It's here: [238c915](https://github.com/anna-teruel/BRAD/commit/238c915213113ed21274b38aa93acba8884cb2cc)

I think we just need to decide how we want to do that. I just completely haven't thought about it at the moment.


> So far, I see that total coverage: 39%, because the Dash app is completely untested. We can leave it as it is for now, but we need to think on how to test the Dash app?
> 
> One option would be to start by testing callback and figure generation functions directly with fixtures and mock inputs, and later add a small number of end-to-end tests that simulate user interactions (button clicks, dropdown selections, file loading, etc.) using Dash's testing framework. What do you think?

TBD I guess we can do that if we are happy with the current way the dashboard is looking

File: tests/test_coord_system.py (hunk 6 above)
Comment (anna-teruel) [reply tests/test_coord_system.py:1]: What type of weirdness? what did you find?

**Comment by anna-teruel** [comment]
> I think we just need to decide how we want to do that. I just completely haven't thought about it at the moment.

What do you think is better?

**Comment by anna-teruel** [comment]
I have problem passing the tests locally on my linux. But I think it's specific to  my chrome. But I think that, for now, we can just merge this. Looks good to me and the coverage is pretty high. We can further discuss more tests in the future.
